Src/Newtonsoft.Json/Serialization/JsonSerializerInternalReader.cs

```diff
@@ -169,9 +169,12 @@ public object Deserialize(JsonReader reader, Type objectType, bool checkAddition
 
                 if (checkAdditionalContent)
                 {
-                    if (reader.Read() && reader.TokenType != JsonToken.Comment)
+                    while (reader.Read())
                     {
-                        throw new JsonSerializationException("Additional text found in JSON string after finishing deserializing object.");
+                        if (reader.TokenType != JsonToken.Comment)
+                        {
+                            throw new JsonSerializationException("Additional text found in JSON string after finishing deserializing object.");
```

Comment:
I wonder if this thrown exception should be changed to include information from the reader via https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/JsonSerializationException.cs#L83